Superlib.js

```diff
@@ -0,0 +1,633 @@
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}function trim(content){return content.replace(/^[\xA0\s]+/gm, '').replace(/[\xA0\s]+$/gm, '').replace(/\n+/g, '\n').replace(/:\n+/g, ': ').replace(/]\n/g, ']').replace(/】\n/g, '】').replace(/\n\/\n/g, '/')}
+
+// https://aurimasv.github.io/z2csl/typeMap.xml#map-book
+// https://aurimasv.github.io/z2csl/typeMap.xml#map-conferencePaper
+// https://aurimasv.github.io/z2csl/typeMap.xml#map-thesis
+// https://aurimasv.github.io/z2csl/typeMap.xml#map-journalArticle
+// https://aurimasv.github.io/z2csl/typeMap.xml#map-patent
+var TYPE_MAP = {
+	图书: 'book',
+	会议论文: 'conferencePaper',
+	学位论文: 'thesis',
+	期刊: 'journalArticle',
+	专利: 'patent'
+};
+
+function doPerson(item, data) {
+	// （日）本田晃一著；朱运程译 translator author
+	if (!data || data.length <= 0) return;
+	const persons = data.split(/；|;/g);
+	for (var person of persons) {
+		if (person.endsWith('著')) {
+			item.creators.push({
+				lastName: person.trim().replace(/著/g, ''),
+				creatorType: 'author',
+				fieldMode: 1
+			});
+		} else if (person.endsWith('译')) {
+			item.creators.push({
+				lastName: person.trim().replace(/译/g, ''),
+				creatorType: 'translator',
+				fieldMode: 1
+			});
+		} else {
+			item.creators.push({
+				lastName: person,
+				creatorType: 'author',
+				fieldMode: 1
+			});
+		}
+	}
+}
+
+function doInventor(item, data) {
+	if (!data || data.length <= 0) return;
+	const persons = data.split(/，/g);
+	for (var person of persons) {
+		item.creators.push({
+			lastName: person,
+			creatorType: 'inventor',
+			fieldMode: 1
+		});
+	}
+}
+
+function doTag(item, data) {
+	if (!data || data.length <= 0) return;
+	const tags = data.split('；');
+	for (var tag of tags) {
+		item.tags.push(tag.trim());
+	}
+}
+
+function sourceType(doc) {
+	var type = doc.querySelector('.on span').textContent;
+	return type;
+}
+
+function detectType(doc) {
+	return TYPE_MAP[sourceType(doc)];
+}
+
+function detectWeb(doc, url) {
+	if (url.includes('/search')) {
+		return getSearchResults(doc, true) ? 'multiple' : false;
+	} else {
+		var dType = detectType(doc, url);
+		if (dType) {
+			return dType;
+		}
+	}
+	
+	return false;
+}
+
+function getSearchResults(doc, checkOnly, itemInfo) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.book1');
+	for (let row of rows) {
+		let a = row.querySelector('td>table .px14');
+		if (!a) {
+			a = row.querySelector('a');
+		}
+		if (!a) {
+			continue;
+		}
+
+		if (checkOnly) return true;
+		
+		let url = a.href;
+
+		// Z.debug(url);
+		let title = ZU.trimInternal(a.textContent);
+		
+		found = true;
+		if (!found['']) {
+			items[''] = '【提醒：存在失败的可能，请隔几个小时后试试，或单个抓取。】';
```

Comment:
Don't really understand this either. You can log an error, maybe, but adding it under an empty key in the object doesn't make sense to me.